src/auth.c

```diff
@@ -312,8 +313,11 @@ auth_client_trust(const char *mac)
 
 	LOCK_CONFIG();
 
-	if (!add_to_trusted_mac_list(mac) && !iptables_trust_mac(mac)) {
-		rc = 0;
+	rc = fw_gops.trust_mac(mac);
+	if (rc) {
```

Comment:
The logic looks inverted here.
iptables_trust_mac() returns 0 on success. So it should if "if (rc == 0)" or "if (!rc)", no?

Reply:
Also probably in some other places.